sycl/include/sycl/detail/common.hpp

```diff
@@ -119,6 +119,72 @@ struct code_location {
 #define _CODELOCFW(a)
 #endif
 
+/// @brief The TLS variable into which Code Locations will be stashed
+extern thread_local detail::code_location GCodeLocTLS;
+
+/// @brief Data type that manages the code_location information in TLS
+/// @details As new SYCL features are added, they all enable the propagation of
+/// the code location information where the SYCL API was called by the
+/// application layer. In order to facilitate this, the tls_code_loc_t object
+/// assists in managing the data in TLS : (1) Populate the information when you
+/// at the top level function in the call chain and (2) to remove the
+/// information when the object goes out scope in the top level function.
```

Comment:
I'd reformat to start each numbered item in new line. Something like

```
some text... data in TLS:
  1) Populate ... text
     more text
      
  2) Remove ...
```

Also, s/when you at the top/when you are at the top/ ?

Reply:
Updated the comment to reflect this request.

Reply:
Have address this with additional documentation that outlines what top of the stack means.